src/pkg.cc

```diff
@@ -57,6 +59,126 @@ UInt T_SEMI   = 0;
 UInt T_BIPART = 0;
 UInt T_BLOCKS = 0;
 
+// Forward decl
+Obj FuncIMAGE_SET_TRANS_INT(Obj, Obj, Obj);
+
+static inline Int IS_MUTABLE_PLIST(Obj list) {
+  return !((TNUM_OBJ(list) - T_PLIST) % 2);
+}
+
+static void REMOVE_DUPS_PLIST_INTOBJ(Obj res) {
+  Obj  tmp;
+  UInt i, k, len;
+  Obj* data;
+
+  len = LEN_PLIST(res);
+
+  if (0 < len) {
+    data = ADDR_OBJ(res);
+    tmp  = data[1];
+    k    = 1;
+    for (i = 2; i <= len; i++) {
+      if (tmp != data[i]) {
+        k++;
+        tmp     = data[i];
+        data[k] = tmp;
+      }
+    }
+    if (k < len) {
+      ResizeBag(res, (k + 1) * sizeof(Obj));
+      SET_LEN_PLIST(res, k);
+    }
+  }
+}
+
+static void SORT_PLIST_CYC(Obj res) {
+  Obj  tmp;
+  UInt h, i, k, len;
+
+  len = LEN_PLIST(res);
+
+  if (0 < len) {
+    h = 1;
+    while (9 * h + 4 < len) {
+      h = 3 * h + 1;
+    }
+    while (0 < h) {
+      for (i = h + 1; i <= len; i++) {
+        tmp = CONST_ADDR_OBJ(res)[i];
+        k   = i;
+        while (h < k && ((Int) tmp < (Int)(CONST_ADDR_OBJ(res)[k - h]))) {
+          ADDR_OBJ(res)[k] = CONST_ADDR_OBJ(res)[k - h];
+          k -= h;
+        }
+        ADDR_OBJ(res)[k] = tmp;
+      }
+      h = h / 3;
+    }
+    CHANGED_BAG(res);
+  }
+}
```

Comment:
If you check for the kernel version as suggested above, then you could also remove this function and just call `SortPlistByRawObj`